commander.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync"
+)
+
+// commander keeps track of a directory and runs commands on them providing
+// a crude form of parallelism.
+type commander struct {
+	*sync.Mutex
+	path string
+	// checkin is a buffered channel. It's length limits the amount of goroutines running commands at once.
+	checkin chan struct{}
+}
+
+func newCommander() commander {
+	path, err := os.Getwd()
+	if err != nil {
+		log.Fatal("getting current dir:", err)
+	}
+	goroutines := runtime.NumCPU() / 2
```

Comment:
Maybe make this a a flag passed in to newCommander instead of querying `runtime`?